app/scrapers/majors_scraper.rb

```diff
@@ -1,45 +1,82 @@
 require 'open-uri'
 require 'nokogiri'
 
-require_relative 'scraper_common.rb'
-include ScraperCommon
+require_relative 'scraper_common'
 
-require_relative '../models/majors.rb'
+require_relative '../models/majors'
 
-prog_name = 'majors_scraper'
+# logger = ScraperCommon.logger
+class MajorsScraper
+  include ScraperCommon
+  attr_accessor :prog_name, :majors_regex
 
-logger = ScraperCommon.logger
+  @@page_url = 'https://admissions.umd.edu/explore/colleges-and-schools/majors/majors-alphabetically'
 
-url = 'https://admissions.umd.edu/explore/colleges-and-schools/majors/majors-alphabetically'
-page = ScraperCommon.get_page url, prog_name
-major_divs = page.css('.page--inner-content a')
-majors = []
-major_divs.each do |link|
-  # parse the name to grab the major's name and its college
+  def self.url
+    @@page_url
+  end
 
-  major_url = link['href']
-  next unless major_url&.include?('umd.edu')
+  def initialize
+    @majors_regex = /(.+)\|(.+)/  # Extracts info from a single major page entry
+    @prog_name = 'majors_scraper' # provide a progname label for logger
+    logger.progname = @prog_name
+  end
 
-  major_parts = /(.+)\|(.+)/.match(link.text)
+  # Scrapes the admissions page for a list of majors
+  #
+  # @param [Nokogiri::HTML::Document] page the HTML page to extract majors from
+  # @return [Array<Major>] a list of majors found on the page
+  def scrape_page(page)
+    major_divs = page.css('.page--inner-content a')
+    majors = []
+    major_divs.each do |link|
+      # parse the name to grab the major's name and its college
 
-  next if major_parts.nil?
+      major_url = link['href']
+      next unless major_url&.include?('umd.edu')
 
-  # 0th match is full string, 1st and 2nd elements are the two matches
-  major_name, major_college = major_parts[1, 3].map(&:strip)
+      major_parts = majors_regex.match(link.text)
 
-  majors << {
-    name: major_name,
-    college: major_college,
-    url: major_url
-  }
-end
+      next if major_parts.nil?
+
+      # 0th match is full string, 1st and 2nd elements are the two matches
+      major_name, major_college = major_parts[1, 3].map(&:strip)
+
+      majors << {
+        name: major_name,
+        college: major_college,
+        url: major_url
+      }
+    end
 
-$DB[:majors].delete
-majors.each do |major|
-  logger.info(prog_name) { "inserting #{major[:name]}" }
+    majors
+  end
 
-  major[:major_id] = major[:name].upcase.gsub!(/[^0-9A-Za-z]/, '')
-  major[:major_id] = major[:name].upcase if major[:major_id].nil?
-  $DB[:majors].insert_ignore.insert(major_id: major[:major_id], name: major[:name], college: major[:college], url: major[:url])
+  # @param [Array<Major>]
+  # @return [void]
+  def update_db(majors)
```

Comment:
Should specify as a comment that this is a delete and replace update rather than an appebd update, or, add a param

Reply:
I added comments just for type checking purposes. Original code is unmodified - just extracted into functions